src/topology/metric_space/infsep.lean

```diff
@@ -0,0 +1,464 @@
+/-
+Copyright (c) 2022 Wrenna Robson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Wrenna Robson
+-/
+import topology.metric_space.basic
+
+/-!
+# Infimum separation
+
+This file defines the extended infimum separation of a set. This is approximately dual to the
+diameter of a set, but where the extended diameter of a set is the supremum of the extended distance
+between elements of the set, the extended infimum separation is the infimum of the (extended)
+distance between *distinct* elements in the set.
+
+We also define the infimum separation as the cast of the extended infimum separation to the reals.
+This is the infimum of the distance between distinct elements of the set when in a pseudometric
+space.
+
+All lemmas and definitions are in the `set` namespace to give access to dot notation.
+
+## Main definitions
+* `set.infesep`: Extended infimum separation of a set.
+* `set.infsep`: Infimum separation of a set (when in a pseudometric space).
+
+!-/
+
+variables {α β : Type*}
+
+namespace set
+
+section infesep
+open_locale ennreal
+open function
+
+/-- The "extended infimum separation" of a set with an edist function. -/
+noncomputable def infesep [has_edist α] (s : set α) : ℝ≥0∞ :=
```

Comment:
Would it make sense to rename this to `einfsep`? That reads better in my mind...